Update contributors on OSCAR website

HereAround opened the pull request:
This PR aims to update the contributor list on the OSCAR site. Contributions to OSCAR.jl, AbstractAlgebra.jl, Singular.jl, Nemo.jl, Hecke.jl, GAP.jl and Polymake.jl are taken into account.

As suggested by @fingolfin, I have classified the contributors as follows:
* PIs,
* Active contributors (that is, they added code during the last year),
* All remaining contributors are marked as "former contributors". 

Here is a preview (two separate screenshots, there is no "cut" in the middle of the actual website):

![Screenshot_20240612_165657](https://github.com/oscar-system/oscar-website/assets/11525279/a9be4499-2d50-401a-8508-6d724d8c878b)
![Screenshot_20240612_165714](https://github.com/oscar-system/oscar-website/assets/11525279/efde1c9e-e8e1-499d-9f76-aa02231868d4)

I must mention that for some people listed on the website, finding their information turned into detective's work, which I hope to have completed correctly. Here are some of the trickiest cases:
* Justus Springer: Cannot connect github account to website at Universität Tübingen.
* Lena Weis: I could not identify her github account. Yet, she contributed code to `OSCAR`. How is this possible?

Despite spending all afternoon on this task, I could not cover all grounds. Rather, the following contributors are yet to be added. (Lists obtained with `git shortlog -s --since=2023-06-12`):

Oscar.jl
     1  Andrei Comaneci
     2  BjSchaefer
     1  Jan Philipp Thiele
     2  Kamillo Ferry
     1  Laura Voggesberger
     1  Marcel Salmon
     1  Nicolas
     3  Tobias Boege

Singular.jl
     1  Albin Ahlbäck
     2  Alexander Demin
     2  Julien Schanz

AbstractAlgebra.jl
(All listed on the website)

Nemo.jl
     3  Albin Ahlbäck
     3  Alexander Demin
     1  Yichao Yu
     1  nanleij

Hecke.jl
     1  Berenike Dieterle
     1  Fabian Gundlach
     1  Grégory Vanuxem
     1  Sreevatsa2
     2  Tobias Braun

Polymake.jl
(All listed on the website)

GAP.jl
(All listed on the website)

I will leave it as such for now. @fingolfin @fieker, what are your thoughts? Are people missing? Is there incomplete/wrong information? Desire for a more restrictive criterion  identifying contributors? Etc.

Once (If?) you are happy, we could merge this and bring this up in the Friday meeting to hear more thoughts.

fingolfin commented:
First off: don't try to hard, it is OK to just list a name without any affiliation or username, those can be added later on if requested, or just left off.

I think something is wrong with your "Singular.jl" list -- I think it is really for AbstractAlgebra.

To find out who e.g. Julien Schanz is you can do this:
1. use `git log --author=Schanz` to find a commit authored by him
2. copy that commit hash 
3. go to https://github.com/Nemocas/AbstractAlgebra.jl/pulls and paste the sha1 you just copied into the "filters" input field (replacing the existing content `is:pr is:open`) and press return
4. now you get a list of all PRs in which this commit originally appeared. from this it is often possible to figure out something about that person, e.g. here that the GitHub user name is `@julien-schanz` (in this case he doesn't have much in his Bio, but I can tell you this is <https://www.uni-saarland.de/lehrstuhl/weber-moritz/team/julien-schanz.html>)


But again: I am not expecting you to do that for everyone. And I'd just drop incomplete names like `nanleij` or `Nicolas`

Some more hints:
- Laura Voggesberger is a former phd student of Gunter Malle from TU Kaiserslautern
- Albin Ahlbäck is a major FLINT contributor
- Berenike Dieterle is a student here at RPTU (worked for @fieker as a HiWi)
- Yichao Yu is a Julia developer (so you can probably find his username at the Julia repo)

fingolfin reviewed (approved):
Anyway from my POV this is fine to merge, it is already a clear improvement and we can add more people later. But I'll leave it to @HereAround to decide when to merge exactly (whatever works best for you: merge it now and improve later, or update the PR first)

fingolfin commented on contributors.md:
```diff
@@ -5,250 +5,432 @@ title: Contributors
   - name: Martin Bies
     affiliation: University of Kaiserslautern-Landau
     website: https://martinbies.github.io
+    email: martin.bies@rptu.de
```
Actually seeing this now: I am not sure we should add email addresses for everyone, this might turn into a privacy / data protection headache, at least if those email addresses are not otherwise public (maybe they all are in the git commit logs, though?)

It should be fine for the PIs. Dunno

HereAround replied:
All emails are public in at least one way. Say by visiting the linked website and then looking up the email there. Once could (in my case) also go through my github profile, which will again link to my website.

Still, if you feel I should remove emails, then this is of course an easy fix.

For all it matters - I have also added emails for all OSCAR book contributors. So this issue repeats for the OSCAR book website/contributors.

fingolfin commented on contributors.md:
```diff
@@ -5,250 +5,432 @@ title: Contributors
+    github: mjrodgers
+
+  - name: Felix Röhrich
+    affiliation: RWTH-Aachen
```
```suggestion
    affiliation: RWTH Aachen University
```

fingolfin commented on contributors.md:
```diff
@@ -269,12 +451,12 @@ contributors:
 <ul>
 {% for p in page.contributors %}
 {% if p.is_active_PI != true %}
+{% if p.is_retired != true %}
```
It might be simpler if you just split the list into three, one for active PIs, one for active contributors, one for former contributors. Then you can remove `is_active_PI` and `is_retired` everywhere

HereAround replied:
I believe you have a point.

HereAround replied:
Done.

fingolfin commented on contributors.md:
```diff
@@ -300,5 +482,44 @@ OSCAR project.
+
+## Former Contributors
+
+The following people did contribute to the OSCAR project in the past, but have not added code since June 2023.
```
```suggestion
The following people did contribute to the OSCAR project in the past, but have not contributed code changes in the past 12 month (as of June 2024).
```
otherwise this will be just factually wrong in short order.


Also, I think we should also somewhere place instructions how to update these lists, i.e. mention running `git shortlog -s --since="1 year ago"` in all our repositories. 

And finally we kinda need to remember to update this regularly. Someone should set a regular calendar reminder maybe? Or even partially automate it via a GitHub action (might be too much work for too little gain, though)

HereAround replied:
I wanted to avoid this issue, hence the hard-coded date. Honestly, I believe we do not want to repeat this exercise once per month or so. Let me ping @aaruni96 so we can gauge if it is feasible to automate the process. If not, I would suggest to keep the hard-coded date. At least for the next month or so.

fingolfin replied:
It would definitely be possible to automate this, but not worth the effort.

But as I see it, your current text does not avoid "the issue", to the contrary: if we don't update the page regularly, then in 6 months it'll suggest that "the following people" did not contribute for the past *18* month. Moreover, if one of them actually makes a contribution next week, we still make this (now incorrect) claim about them.

While with my version, it is clear that (a) this is about "contribution of the past 12 month" (so the next time someone wants to update it, they know what the cut-off point is), and it also makes it clearly that the list is current as of June 2024 -- so someone reading it in December can take this into account "ah, so that's why I am listed under 'former contributor' -- guess I should ping them to update this page"

HereAround replied:
Sorry my bad. Your suggestion is perfect. I just had to read carefully. Has been applied.

aaruni96 replied:
I'm sure I can come up with some convoluted way to automatically update this, but its not going to be quickly.

Its trivial to ask for "date one year ago" (`date -d "last year" +"%Y-%m-%d"`), but less clear what to do with the the output of git shortlog, and how to programmatically modify an md file.

HereAround replied:
Let us gauge in a month or two (presumably someone - maybe I) has to repeat this on a regular (monthly?) basis. If this process turns out to be time consuming each time, then we can investigate further.

HereAround commented:
Thank you @fingolfin for the many ideas. Let me address them in this PR. Will ping you once I made those changes.

HereAround committed:
Update contributors

Co-authored-by: Max Horn <max@quendi.de>

fieker commented:
Pi's: Simon und Tommy fehlen

On Wed, 12 Jun 2024, 19:51 Martin Bies, ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In contributors.md
> <https://github.com/oscar-system/oscar-website/pull/344#discussion_r1636876850>
> :
>
> >      github: alexjbest
>
>    - name: Martin Bies
>      affiliation: University of Kaiserslautern-Landau
>      website: https://martinbies.github.io
> +    email: ***@***.***
>
> All emails are public in at least one way. Say by visiting the linked
> website and then looking up the email there. Once could (in my case) also
> go through my github profile, which will again link to my website.
>
> Still, if you feel I should remove emails, then this is of course an easy
> fix.
>
> For all it matters - I have also added emails for all OSCAR book
> contributors. So this issue repeats for the OSCAR book website/contributors.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/oscar-system/oscar-website/pull/344#discussion_r1636876850>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA36CV33P5R3ESTFDDA5MEDZHCDB3AVCNFSM6AAAAABJGS4TRWVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDCMJTG4YDSMZYGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

HereAround commented:
> Pi's: Simon und Tommy fehlen
> […](#)
> On Wed, 12 Jun 2024, 19:51 Martin Bies, ***@***.***> wrote: ***@***.**** commented on this pull request. ------------------------------ In contributors.md <[#344 (comment)](https://github.com/oscar-system/oscar-website/pull/344#discussion_r1636876850)> : > github: alexjbest - name: Martin Bies affiliation: University of Kaiserslautern-Landau website: https://martinbies.github.io + email: ***@***.*** All emails are public in at least one way. Say by visiting the linked website and then looking up the email there. Once could (in my case) also go through my github profile, which will again link to my website. Still, if you feel I should remove emails, then this is of course an easy fix. For all it matters - I have also added emails for all OSCAR book contributors. So this issue repeats for the OSCAR book website/contributors. — Reply to this email directly, view it on GitHub <[#344 (comment)](https://github.com/oscar-system/oscar-website/pull/344#discussion_r1636876850)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AA36CV33P5R3ESTFDDA5MEDZHCDB3AVCNFSM6AAAAABJGS4TRWVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDCMJTG4YDSMZYGI> . You are receiving this because you were mentioned.Message ID: ***@***.***>

Thank you. Modified accordingly.

HereAround commented:
All changes done except for:
* Email addresses are still in for mostly everybody (unless I could not find them). They are public in all cases, I believe.
* I have not changed the text detailing "Former Contributors", because I want to avoid the automation issue/repeat this task once every month. Unless of course e.g. @aaruni96 tells me that the automation can be set up super quick.

With that being said, @fingolfin @fieker please consider merging if you are happy with the changes.

HereAround commented:
(Oh, and I have gone over the remaining names in the list above and added details on these people.)

fingolfin commented on contributors.md:
```diff
@@ -1,280 +1,498 @@
+    github: sumiya11
+
+  - name: Berenike Dieterle
+    github: BD-00
```
```suggestion
    affiliation: University of Kaiserslautern-Landau
    github: BD-00
```

HereAround committed:
Update contributors.md

Co-authored-by: Max Horn <max@quendi.de>

benlorenz commented on contributors.md:
```diff
@@ -1,280 +1,499 @@
-    affiliation: RWTH Aachen University
-    website: http://www.math.rwth-aachen.de/~Frank.Luebeck/index.html
+    website: https://www.tu.berlin/dmg/ueber-uns/systemadministration
+    email: benjamin.lorenz@tu-berlin.de
```
```suggestion
    email: lorenz@math.tu-berlin.de
```

benlorenz commented on contributors.md:
```diff
@@ -1,280 +1,499 @@
+  - name: Lena Weis
+    affiliation: TU Berlin
+    website: https://page.math.tu-berlin.de/~weis/
+    email: weis@math.tu-berlin.de
```
```suggestion
    email: weis@math.tu-berlin.de
    github: lenweis
```

HereAround replied:
Cool!

I have been trying to find this username and failed. For instance, the one you mention (I take it to be correct) is not listed on the contributors page: https://github.com/oscar-system/Oscar.jl/graphs/contributors. Do you know why?

HereAround replied:
Or am I just too stupid to see it?

HereAround replied:
(I am just curious.)

benlorenz replied:
> Cool!
> 
> I have been trying to find this username and failed. For instance, the one you mention (I take it to be correct) is not listed on the contributors page: [oscar-system/Oscar.jl/graphs/contributors](https://github.com/oscar-system/Oscar.jl/graphs/contributors). Do you know why?

Maybe because the email that was used in the commits in not listed in the her github settings.
But she is listed in the organization members list: https://github.com/orgs/oscar-system/people?query=lena

HereAround committed:
Update contributors.md

Co-authored-by: Max Horn <max@quendi.de>

HereAround committed:
Update contributors.md

Co-authored-by: Benjamin Lorenz <benlorenz@users.noreply.github.com>

HereAround committed:
Update contributors.md

Co-authored-by: Benjamin Lorenz <benlorenz@users.noreply.github.com>

HereAround commented:
@fieker and @fingolfin , I believe this is ready for the next round of review.

fieker reviewed (approved):
Thanks
